OloEditor/src/Panels/TilemapPainterPanel.cpp

```
@@ -0,0 +1,510 @@
+
+        const f32 t = -localOrigin.z / localDir.z;
+        if (!std::isfinite(t) || t < 0.0f)
+            return false;
+
+        const glm::vec3 local = localOrigin + t * localDir;
+        if (!std::isfinite(local.x) || !std::isfinite(local.y))
+            return false;
+
+        const f32 tileSize = tilemap.TileSize;
+        if (local.x < 0.0f || local.y < 0.0f)
+            return false;
```

Cppcheck (2.21.0)

[style] The function 'CanUndo' is never used.

(unusedFunction)